src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,64 @@
 package mate.academy;
 
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    public static final int THREAD_POOL_SIZE = 1;
+    public static final int SHUTDOWN_TIMEOUT_MINUTES = 2;
+    private final ThreadPoolExecutor executor;
+    private final List<EventListener> listeners;
 
+    public EventManager() {
+        this.executor = new ThreadPoolExecutor(
+                THREAD_POOL_SIZE,
+                THREAD_POOL_SIZE,
+                0L,
+                TimeUnit.MILLISECONDS,
+                new LinkedBlockingQueue<>()
+        );
+        this.listeners = new CopyOnWriteArrayList<>();
     }
 
-    public void deregisterListener(EventListener listener) {
+    public void registerListener(EventListener listener) {
+        listeners.add(listener);
+        setUpNewThreadPoolSize();
+    }
 
+    public void deregisterListener(EventListener listener) {
+        listeners.remove(listener);
+        setUpNewThreadPoolSize();
     }
 
     public void notifyEvent(Event event) {
```

Comment:
The notifyEvent method submits tasks to the executor within a loop. Be aware that if the listeners list is large, this could lead to a large number of tasks being queued simultaneously. If this is a concern, consider strategies to limit the rate of task submission or to handle back pressure.